packages/map-template/src/components/LegendDialog/LegendDialog.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useRef, useState } from "react";
+import './LegendDialog.scss';
+import { useRecoilState, useRecoilValue } from "recoil";
+import primaryColorState from "../../atoms/primaryColorState";
+import isLegendDialogVisibleState from "../../atoms/isLegendDialogVisibleState";
+import { useTranslation } from "react-i18next";
+import { useIsKioskContext } from "../../hooks/useIsKioskContext";
+import legendSortedFieldsSelector from "../../selectors/legendSortedFieldsSelector";
+
+/**
+ * Handle the Legend dialog.
+ *
+ */
+function LegendDialog() {
+    const { t } = useTranslation();
+    const primaryColorProp = useRecoilValue(primaryColorState);
+
+    const legendSections = useRecoilValue(legendSortedFieldsSelector);
+
+    const isKioskContext = useIsKioskContext();
+
+    const legendModalRef = useRef();
+
+    const legendSectionRef = useRef();
+
+    const scrollButtonsRef = useRef();
+
+    const [showLegendDialog, setShowLegendDialog] = useRecoilState(isLegendDialogVisibleState);
+
+    const [isLegendScrollable, setIsLegendScrollable] = useState(false);
+
+    const [showScrollButtons, setShowScrollButtons] = useState(false);
+
+    const legendSectionsRef = useRef();
+
+    /*
+     * Get the height of the legend sections. 
+     * Set the legend size atom and determine 
+     * if the scroll buttons should be shown.
+     */
+    useEffect(() => {
+        if (showLegendDialog && isKioskContext) {
+            if (legendSectionRef.current.clientHeight > 700) {
+                setIsLegendScrollable(true);
+                setShowScrollButtons(true);
+            } else {
+                setShowScrollButtons(false);
+            }
```

Comment:
Can the `isLegendScrollable` and `showScrollButtons` be merged into one? They seem to have the same values.

Reply:
That's true, great idea! ✅